runtime/src/accounts_db.rs

```diff
@@ -1032,14 +1044,17 @@ impl AccountsDB {
                         if forced { " (forced)" } else { "" },
                     );
                     return 0;
-                } else if (alive_count as f32 / stored_count as f32) >= 0.80 && !forced {
-                    trace!(
-                        "shrink_stale_slot ({}): not enough space to shrink: {} / {}",
-                        slot,
-                        alive_count,
-                        stored_count,
+                } else if !forced {
+                    let sparse_by_count = (alive_count as f32 / stored_count as f32) <= 0.8;
+                    let sparse_by_bytes = (written_bytes as f32 / total_bytes as f32) <= 0.8;
```

Comment:
the reason I intentionally didn't this at the first implementation is that I didn't care about unused mmaps. But, well, it turned out address space is scarce resource indeed... ;)